### PR TITLE
LUCENE-10289: Change DocIdSetBuilder#grow() from taking an int to a long

### PR DESCRIPTION
**Pull request by iverase**
It makes sense as it is possible to bulk add more than Integer.MAX_VALUE docs as there can be duplicates.

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -100,6 +100,8 @@ Other
 
 * LUCENE-10284: Upgrade morfologik-stemming to 2.1.8. (Dawid Weiss)
 
+LUCENE-10289: DocIdSetBuilder#grow() takes now a long instead of an int. (Ignacio Vera)
```
Comment (jpountz) [inline lucene/CHANGES.txt:103]:
```suggestion
 * LUCENE-10289: DocIdSetBuilder#grow() takes now a long instead of an int. (Ignacio Vera)
```

Comment (jpountz) [reply lucene/CHANGES.txt:103]: Also, move the CHANGES entry under `API changes`?

File: lucene/core/src/java/org/apache/lucene/util/DocIdSetBuilder.java [hunk 2]
```diff
@@ -184,10 +184,11 @@ public void add(DocIdSetIterator iter) throws IOException {
    * Reserve space and return a {@link BulkAdder} object that can be used to add up to {@code
    * numDocs} documents.
    */
-  public BulkAdder grow(int numDocs) {
+  public BulkAdder grow(long numDocs) {
     if (bitSet == null) {
       if ((long) totalAllocated + numDocs <= threshold) {
-        ensureBufferCapacity(numDocs);
+        // threshold is an int, cast is safe
+        ensureBufferCapacity((int) numDocs);
```
Comment (jpountz) [inline lucene/core/src/java/org/apache/lucene/util/DocIdSetBuilder.java:191]: can you still use `Math.toIntExact` instead for safety?

**Review by jpountz — APPROVED** [review]

File: lucene/core/src/java/org/apache/lucene/util/DocIdSetBuilder.java (hunk 2 above)
Comment (iverase) [reply lucene/core/src/java/org/apache/lucene/util/DocIdSetBuilder.java:191]: Ok, out of paranoia I have added checks for long Overflow too

File: lucene/CHANGES.txt (hunk 1 above)
Comment (iverase) [reply lucene/CHANGES.txt:103]: done

**Pull request merged**